### ISSUE TITLE
Uncaught TypeError: Cannot read property 'get' of null

### ISSUE THREAD

**Issue by nsimmonds** [rancher/rancher#9727]
*Describe your issue here*

I'm encountering an error in rancher/server where one of my stacks throws an error when trying to view it in the GUI.  The error is in this function:

```
displayDetail: (function () {
      var out = '';
      if (this.get('hostname')) {
        out = esc(this.get('hostname'));
      } else {
        var ips = this.get('externalIpAddresses');
        var num = ips.get('length');
        for (var i = 0; i < 3 && i < num; i++) {
          out += '<span>' + (i === 0 ? '' : ', ') + esc(ips.objectAt(i)) + '</span>';
        }

        if (num > 3) {
          out += ' and ' + (num - 3) + ' more';
        }
      }

      return ('<span class="text-muted">To: </span>' + out).htmlSafe();
    }

```
At `var num = ips.get('length');`

We're using hostnames. It looks like, in that case, it should just skip the gathering of IPs and therefore shouldn't error. Somehow it's getting into the `else` block anyway.

**Update:** someone had added an external service to docker-compose.yml, and also rancher-compose.yml, but they didn't give it either a hostname or IP addresses. We've fixed it, but it seems like a) the CLI should have thrown an error for bad config and b) absent that, a single Javascript error shouldn't choke the UI display of the entire stack.

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.6.7` Cattle: `v0.183.5` UI: `v1.6.17` |
|Access|`github` `admin`|
|Orchestration|`Cattle`|
|Route|`stack.index`|

**Comment by Tzrlk** [comment]
You can temporarily fix it by shelling into your rancher container, going to `/usr/share/cattle/${hashcode}/assets` and editing `ui-${hashcode}` to add `||[]` immediately after any occurrence of `this.get("externalOpAddresses")`. Can probably do it with `sed`, but I haven't needed to reproduce the fix yet.

**Comment by Tzrlk** [comment]
However, it looks as though this is fixed in master: https://github.com/rancher/ui/blob/master/app/models/externalservice.js

**Comment by superseb** [comment]
Master is used for `2.0`, can you give a reproducible case to get this fixed?

**Comment by gaetancollaud** [comment]
I have the same issue when I use variables in rancher-compose.yml. When the IPs where fixed it worked
```
.catalog:
  name: myproject
  version: 40.0
  description: My project
  questions:
    - variable: core_ip
      label: Core ip address
      type: string
      default: 1.2.3.4
      required: true
  services:
    core:
      external_ips:
      - ${core_ip}
      start_on_create: true
...
```

As you can see on the right, `externalIpAddresses` is null
![capture](https://user-images.githubusercontent.com/3659220/35042170-c307f00a-fb87-11e7-8d70-5f8ff25785d0.PNG)

**Comment by maltyxx** [comment]
me too

**Comment by onmomo** [comment]
@Tzrlk Thx a lot! Was able to fix it with the following:

```
docker exec -it ${rancher-server-container} /bin/bash
cd /usr/share/cattle/${hashcode}/assets
sed -i 's/this.get("externalIpAddresses")/this.get("externalIpAddresses")||[]/g' ui-${another-hashcode}.js 
```

**Comment by loganhz** [comment]
Fixed in v1.6.22

Please let us know in a comment if you can still reproduce the issue, and we'll reopen it
